safekeeper/src/timeline.rs

```diff
@@ -129,15 +136,62 @@ impl SharedState {
             notified_commit_lsn: Lsn(0),
             sk: SafeKeeper::new(zttid.timeline_id, control_store, wal_store, conf.my_id)?,
             replicas: Vec::new(),
+            wal_backup_active: false,
             active: false,
             num_computes: 0,
             pageserver_connstr: None,
             last_removed_segno: 0,
         })
     }
+    fn is_active(&self) -> bool {
+        self.is_wal_backup_required()
+            // FIXME: add tracking of relevant pageservers and check them here individually,
+            // otherwise migration won't work (we suspend too early).
+            || self.sk.inmem.remote_consistent_lsn <= self.sk.inmem.commit_lsn
+    }
+
+    /// Mark timeline active/inactive and return whether s3 offloading requires
+    /// start/stop action.
+    fn update_status(&mut self) -> bool {
+        self.active = self.is_active();
+        self.is_wal_backup_action_pending()
+    }
+
+    /// Should we run s3 offloading in current state?
+    fn is_wal_backup_required(&self) -> bool {
+        let seg_size = self.get_wal_seg_size();
+        self.num_computes > 0 ||
+        // Currently only the whole segment is offloaded, so compare segment numbers.
+               (self.sk.inmem.commit_lsn.segment_number(seg_size) >
```

Comment:
No. If e.g. commit_lsn is `0/2696628`, but backup lsn is `0/2000000`, there is nothing to do currently. We could compare LSNs, but truncate commit_lsn to segment beginning beforehand, not sure that would be simpler.

Reply:
Maybe I don't get something, but due to

```rust
impl Lsn {
// ...
    pub fn segment_number(self, seg_sz: usize) -> u64 {
        self.0 / seg_sz as u64
    }
// ...
}
```

`(self.sk.inmem.commit_lsn.segment_number(seg_size) > self.sk.inmem.backup_lsn.segment_number(seg_size))`
can be pseudo rewritten as 
`(self.sk.inmem.commit_lsn / seg_size) > (self.sk.inmem.backup_lsn / seg_size)`

and since `seg_size` is `usize` and we're sure it's not 0, it's just 

`self.sk.inmem.commit_lsn > self.sk.inmem.backup_lsn`

?

Reply:
Well, in math world, yes. Here, where remainder is thrown away, no.